src/cbc_sdk/workload/compliance_assessment.py

```diff
@@ -0,0 +1,599 @@
+
+        Example:
+            To reassess an object:
+            benchmark_sets = cb.select(ComplianceBenchmark)
+            benchmark_sets[0].execute_action('REASSESS')
+        """
+        ACTIONS = ('ENABLE', 'DISABLE', 'REASSESS')
+
+        if action.upper() not in ACTIONS:
+            raise ApiError("Action is not supported. Options: ENABLE, DISABLE, REASSESS")
+
+        args = {"action": action.upper()}
+        url = ""
+        if device_ids:
+            args['device_ids'] = [device_ids] if isinstance(device_ids, str) else device_ids
+            url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/compliance/device_actions"
+        else:
+            url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/actions"
+
+        return self._cb.post_object(url, body=args).json()
+
+    # API Not supported
+    #
+    # def get_device_compliances(self, query=""):
+    #     """
+    #     Fetches devices compliance summaries associated with the benchmark set.
+    #
+    #     Args:
+    #         query (str, optional): The query to filter results.
+    #
+    #     Returns:
+    #         [dict]: List of Device Compliances
+    #
+    #     Example:
+    #         >>> cb = CBCloudAPI(profile="example_profile")
+    #         >>> benchmark_set = cb.select(ComplianceBenchmark).first()
+    #         >>> rules = benchmark_set.get_device_compliance()
+    #     """
+    #     url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/compliance/devices/_search"
+    #     current = 0
+    #     device_compliances = []
+    #     while True:
+    #         resp = self._cb.post_object(url, body={
+    #             "query": query,
+    #             "rows": 10000,
+    #             "start": current,
+    #             "sort": [
+    #                 {
+    #                     "field": "device_name",
+    #                     "order": "DESC"
+    #                 }
+    #             ]
+    #         })
+    #         result = resp.json()
+    #
+    #         device_compliances.extend(result.get("results", []))
+    #         current = len(device_compliances)
+    #
+    #         if current >= result["num_found"]:
+    #             break
+    #
+    #     return device_compliances
+
+    def get_rule_compliances(self, query=""):
+        """
+        Fetches rule compliance summaries associated with the benchmark set.
+
+        Args:
+            query (str, optional): The query to filter results.
+
+        Returns:
+            [dict]: List of Rule Compliances
+
+        Example:
+            >>> cb = CBCloudAPI(profile="example_profile")
+            >>> benchmark_set = cb.select(ComplianceBenchmark).first()
+            >>> rules = benchmark_set.get_rule_compliance()
+        """
+        url = self.urlobject.format(self._cb.credentials.org_key) + f"{self.id}/compliance/rules/_search"
+        current = 0
+        rule_compliances = []
+        while True:
```

Comment:
Seems like most of the code for getting the rules for specific device or benchmark set is the same, can we extract it to a non-public function to reuse?

Reply:
They are very similar but the sorts are different too so I'd rather keep them separate given the code is simple and in case we want/need to adjust the functionality further in the future